Amicable Number Finder

SarahLammert opened the pull request:
Please check it out. There are two systems. The first one is really slow for large numbers because it finds the sum of factors slowly. The bottom one runs super fast.

SarahLammert committed:
Amicable Number Finder

carlguo866 commented on src/amicableNumbers/AmicableNums.java:
```diff
@@ -0,0 +1,85 @@
+		f2 = System.currentTimeMillis();
+		System.out.println(checked2);
+		System.out.println("All Done");
+		System.out.println("The total time was " + (f2-s2) + " milliseconds");
```
awesome implementation of the time/efficiency check

carlguo866 commented on src/amicableNumbers/AmicableNums.java:
```diff
@@ -0,0 +1,85 @@
+		for(int x = 2; x <= LIMIT; x++) {
+			int p = sumOfFactorsFastVersion(x);
+			if(p <= LIMIT && x != p) {
+				if(x == sumOfFactorsFastVersion(p)) {
```
u can just combine `p <= LIMIT && x != p` and `x == sumOfFactorsFastVersion(p)`

carlguo866 commented on src/amicableNumbers/AmicableNums.java:
```diff
@@ -0,0 +1,85 @@
+			int p = sumOfFactorsFastVersion(x);
+			if(p <= LIMIT && x != p) {
+				if(x == sumOfFactorsFastVersion(p)) {
+					checked2.add(x);
```
put `p` in there as well
also you would want to check if `checked2.contains(x) || checked2.contains(x)` 
In this scenario, if you print every single loop out with sysout, it would appear like this:
220 284
284 220 
1184 1210
1210 1184
which doubles your time

carlguo866 commented on src/amicableNumbers/AmicableNums.java:
```diff
@@ -0,0 +1,85 @@
+		s2 = System.currentTimeMillis();
+		checked2 = new ArrayList<Integer>();
+		
+		for(int x = 2; x <= LIMIT; x++) {
```
good implementation that works

carlguo866 commented on src/test/Test.java:
```diff
@@ -0,0 +1,14 @@
+		ArrayList<String> b = new ArrayList<String>();
+		b.add(s);
+		b.remove(0);
+		System.out.println(s);
```
TODO: Finish the `void removeDuplicate()` function, which is also a part of the homework!

carlguo866 commented on src/amicableNumbers/AmicableNums.java:
```diff
@@ -0,0 +1,85 @@
+	    }
+	    prod -= a;
+	    return prod;
+	}
```
Interesting. I believe it works because my computer says so, but the readability is low. Explain it to me by commenting or tmr in class

rochebox replied:
Hey great you found this code to try, but if you find code you should add a comment that explains where you got it.  Also like Carl said in class the variable names, don't really give any clue about what is going on.... so you would have to modify that (giving attribution to the original author)

SarahLammert committed:
Revert "Amicable Number Finder"

This reverts commit ba6a3c6cabedd2eecf28085e6ee0f7a75e45e1b5.

SarahLammert committed:
Revert "Revert "Amicable Number Finder""

This reverts commit e8de7e73b695c0cc6ee3b14e78da460029b66c31.

SarahLammert committed:
Fixing Github

SarahLammert committed:
Fixing this

SarahLammert committed:
Trying to get my amicable nums on this branch

SarahLammert committed:
Found my files... don't need this copy.

SarahLammert committed:
Commit to match master

carlguo866 commented:
did you do any changes on this branch that you want me to look at it yet?

SarahLammert committed:
adding some guide to amicableNums

Don't understand how the program implements the math, but I do understand what the math is doing to an  extent

SarahLammert committed:
Merge branch 'master' into amicableNums
